rdtools/soiling.py

```diff
@@ -701,3 +704,176 @@ def soiling_srr(energy_normalized_daily, insolation_daily, reps=1000,
         max_negative_step=max_negative_step)
 
     return sr, sr_ci, soiling_info
+
+
+def _count_month_days(start, end):
+    '''Return a dict of number of days between start and end (inclusive) in each month'''
+    days = pd.date_range(start, end)
+    months = [x.month for x in days]
+    out_dict = {}
+    for month in range(1, 13):
+        out_dict[month] = months.count(month)
+
+    return out_dict
+
+
+def annual_soiling_ratios(stochastic_soiling_profiles, confidence_level=68.2):
+    '''
+    Return annualized soiling ratios and associated confidence intervals based
+    on stochastic soiling profiles from SRR. Note that each year may be affected
+    by previous years' profiles for all SRR cleaning assumptions (i.e. method) except
+    perfect_clean.
+
+    Parameters
+    ----------
+    stochastic_soiling_profiles : list
+        List of pd.Series representing profile realizations from the SRR monte carlo.
+        Typically ``soiling_interval_summary['stochastic_soiling_profiles']`` obtained with
+        :py:func:`rdtools.soiling.soiling_srr` or :py:meth:`rdtools.soiling.SRRAnalysis.run`
+
+    confidence_level : float, default 68.2
+        The size of the confidence interval to use in determining the upper and lower
+        quantiles reported in the returned DataFrame. (The median is always included in
+        the result.)
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame describing annual soiling rates.
+        Columns:
+
+        * `year`
+        * `soiling_ratio_median` - The median insolation-weighted soiling ratio for the year
+        * `soiling_ratio_low` - The lower edge of the confidence interval for
+          insolation-weighted soiling ratio for the year
+        * `soiling_ratio_high` - The upper edge of the confidence interval for
+          insolation-weighted soiling ratio for the year
+    '''
+
+    df = pd.DataFrame(pd.concat(stochastic_soiling_profiles))
+
+    ci_quantiles = [0.5 - confidence_level/2/100, 0.5 + confidence_level/2/100]
+    annual_soiling = df.groupby(df.index.year).quantile([0.5, ci_quantiles[0], ci_quantiles[1]])
+    annual_soiling.index.rename(['year', 'quantile'], inplace=True)
+    annual_soiling.columns = ['insolation_weighted_soiling_ratio']
+
+    # Make the output look like monthly_soiling_rates
+    annual_soiling = annual_soiling.reset_index(['quantile'])
+    annual_soiling = pd.pivot(annual_soiling, columns='quantile')
+    annual_soiling.columns = annual_soiling.columns.droplevel(0)
+    annual_soiling.columns.name = None
+    annual_soiling = annual_soiling.sort_index(axis=1)
+    annual_soiling.columns = ['soiling_ratio_low', 'soiling_ratio_median', 'soiling_ratio_high']
+    annual_soiling = annual_soiling[['soiling_ratio_median', 'soiling_ratio_low', 'soiling_ratio_high']]
+    annual_soiling = annual_soiling.reset_index()
```

Comment:
I generally try to avoid monkeying around with multi-indexing and pivoting where possible.  In this case it's because calling `quantile` with a vector argument returns a multi-indexed dataframe, so would it be more straightforward to call quantile three times with scalar arguments?

```python
def annual_soiling_ratios(stochastic_soiling_profiles, confidence_level=68.2):
    all_profiles = pd.concat(stochastic_soiling_profiles)
    annual_groups = all_profiles.groupby(all_profiles.index.year)
    annual_soiling = pd.DataFrame({
        'soiling_ratio_median': annual_groups.quantile(0.5),
        'soiling_ratio_low': annual_groups.quantile(0.5 - confidence_level/2/100),
        'soiling_ratio_high': annual_groups.quantile(0.5 + confidence_level/2/100),
    })
    annual_soiling.index.name = 'year'
    annual_soiling = annual_soiling.reset_index()
    return annual_soiling
```

Reply:
Agreed, I like your much simpler implementation